Sources/Shared/LocalAuthentication/AuthenticationService.swift

```diff
@@ -0,0 +1,49 @@
+import Foundation
+import LocalAuthentication
+import UIKit
+
+public protocol AuthenticationServiceProtocol {
+    var delegate: AuthenticationServiceDelegate? { get set }
+    func authenticate()
+    func invalidate()
+}
+
+public protocol AuthenticationServiceDelegate: AnyObject {
+    func didFinishAuthentication(authorized: Bool)
+}
+
+class AuthenticationService: AuthenticationServiceProtocol {
+    private let context = LAContext()
+
+    weak var delegate: AuthenticationServiceDelegate?
+
+    func authenticate() {
+        var error: NSError?
+        if context.canEvaluatePolicy(.deviceOwnerAuthenticationWithBiometrics, error: &error) {
+            authenticate(policy: .deviceOwnerAuthenticationWithBiometrics)
+        } else if context.canEvaluatePolicy(.deviceOwnerAuthentication, error: &error) {
+            authenticate(policy: .deviceOwnerAuthentication)
+        } else {
+            Current.Log
+                .error(
+                    [
+                        "Failed to authenticate with biometrics": "User context can't evaluate policy for device ownership",
+                    ]
+                )
```

Comment:
what should we tell the user if they fail to do this? should we have on-screen ui that says an error?

Reply:
If the user can't authenticate neither with biometrics or password then the user is locked out, I would say this is very edge case since it's credentials are the same as the iOS/macOS device. Any suggestion?